packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -469,46 +469,108 @@ abstract class SchedulerBinding extends BindingBase {
+
+    String debugBanner;
     assert(() {
-      if (debugPrintBeginFrameBanner)
-        debugPrint('━━━━━━━┫ Begin Frame ($timeStamp) ┣━━━━━━━');
+      _debugFrameNumber += 1;
+      if (debugPrintBeginFrameBanner || debugPrintEndFrameBanner) {
+        StringBuffer frameTimeStampDescription = new StringBuffer();
+        if (rawTimeStamp != null) {
+          if (_currentFrameTimeStamp.inDays > 0)
+            frameTimeStampDescription.write('${_currentFrameTimeStamp.inDays}d ');
+          if (_currentFrameTimeStamp.inHours > 0)
+            frameTimeStampDescription.write('${_currentFrameTimeStamp.inHours - _currentFrameTimeStamp.inDays * Duration.HOURS_PER_DAY}h ');
+          if (_currentFrameTimeStamp.inMinutes > 0)
+            frameTimeStampDescription.write('${_currentFrameTimeStamp.inMinutes - _currentFrameTimeStamp.inHours * Duration.MINUTES_PER_HOUR}m ');
+          if (_currentFrameTimeStamp.inSeconds > 0)
+            frameTimeStampDescription.write('${_currentFrameTimeStamp.inSeconds - _currentFrameTimeStamp.inMinutes * Duration.SECONDS_PER_MINUTE}s ');
+          frameTimeStampDescription.write('${_currentFrameTimeStamp.inMilliseconds - _currentFrameTimeStamp.inSeconds * Duration.MILLISECONDS_PER_SECOND}');
+          int microseconds = _currentFrameTimeStamp.inMicroseconds - _currentFrameTimeStamp.inMilliseconds * Duration.MICROSECONDS_PER_MILLISECOND;
+          if (microseconds > 0)
+            frameTimeStampDescription.write('.${microseconds.toString().padLeft(3, "0")}');
+          frameTimeStampDescription.write('ms');
+        } else {
+          frameTimeStampDescription.write('(warm-up frame)');
+        }
+        debugBanner = '▄▄▄▄▄▄▄▄ Frame ${_debugFrameNumber.toString().padRight(7)}   ${frameTimeStampDescription.toString().padLeft(18)} ▄▄▄▄▄▄▄▄';
+        if (debugPrintBeginFrameBanner)
+          debugPrint(debugBanner);
+      }
       return true;
     });
-    _lastRawTimeStamp = rawTimeStamp;
+
     assert(!_isProducingFrame);
     _isProducingFrame = true;
     _hasScheduledFrame = false;
-    _invokeTransientFrameCallbacks(timeStamp);
-
-    for (FrameCallback callback in _persistentCallbacks)
-      _invokeFrameCallback(callback, timeStamp);
-
-    _isProducingFrame = false;
-
-    List<FrameCallback> localPostFrameCallbacks =
-        new List<FrameCallback>.from(_postFrameCallbacks);
-    _postFrameCallbacks.clear();
-    for (FrameCallback callback in localPostFrameCallbacks)
-      _invokeFrameCallback(callback, timeStamp);
-
-    Timeline.finishSync();
+    try {
 
-    assert(() {
-      if (debugPrintEndFrameBanner)
-        debugPrint('━━━━━━━┫ End of Frame ┣━━━━━━━');
-      return true;
-    });
+      // TRANSIENT FRAME CALLBACKS
+      _invokeTransientFrameCallbacks(_currentFrameTimeStamp);
+
+      // PERSISTENT FRAME CALLBACKS
+      for (FrameCallback callback in _persistentCallbacks)
+        _invokeFrameCallback(callback, _currentFrameTimeStamp);
+      _isProducingFrame = false;
```

Comment:
done, though arguably we should remove this whole try-finally since everything interesting done by this block is itself wrapped in try-catch blocks.